modules/govuk_scripts/templates/govuk_node_list_aws.erb

```diff
@@ -46,6 +46,13 @@ def ec2_nodes(stackname, nodeclass=None):
 
     return(hosts)
 
+def get_hostname_by_id(instance_id):
+    client = boto3.client('ec2', region_name=region)
+    response = client.describe_instances(InstanceIds=[instance_id])
+    hostname = response['Reservations'][0]['Instances'][0]['NetworkInterfaces'][0]['PrivateDnsName']
```

Comment:
You should add handling for when the `PrivateDnsName` isn't present. It's absent by default in a VPC.

Reply:
I've updated this to return `None`. There may be edge cases where InstanceIDs exist in PuppetDB, but the instance has been destroyed. In that case, we wish to leave it out of the result, but not necessarily error.